cmd/manager/main.go

```diff
@@ -51,6 +51,12 @@ func main() {
 		log.Fatalf("failed to get watch namespace: %v", err)
 	}
 
+	// Report that the operator is progressing towards a new version.
+	err = operatorstatus.StartReporting()
```

Comment:
This should only kick off if the API is present. There is no reason for these threads on vanilla kube. But you do need the check in `SendEventMessage()` either way.